numpy/core/tests/test_multiarray.py

```diff
@@ -1059,34 +1059,62 @@ def test_array_too_big(self):
             assert_raises(ValueError, np.ndarray, buffer=buf, strides=(0,),
                           shape=(max_bytes//itemsize + 1,), dtype=dtype)
 
-    def test_jagged_ndim_object(self):
-        # Lists of mismatching depths are treated as object arrays
-        a = np.array([[1], 2, 3])
-        assert_equal(a.shape, (3,))
-        assert_equal(a.dtype, object)
+    def _ragged_creation(self, seq, dtype):
+        # without dtype=object, the ragged object should raise
+        with assert_warns(DeprecationWarning):
+            a = np.array(seq)
+        b = np.array(seq, dtype=object)
+        assert_equal(a, b)
+        return b
 
-        a = np.array([1, [2], 3])
+    @pytest.mark.parametrize('dtype', [object, None])
+    def test_ragged_ndim_object(self, dtype):
+        # Lists of mismatching depths are treated as object arrays
+        a = self._ragged_creation([[1], 2, 3], dtype)
         assert_equal(a.shape, (3,))
         assert_equal(a.dtype, object)
 
-        a = np.array([1, 2, [3]])
+        a = self._ragged_creation([1, [2], 3], dtype)
         assert_equal(a.shape, (3,))
         assert_equal(a.dtype, object)
 
-    def test_jagged_shape_object(self):
-        # The jagged dimension of a list is turned into an object array
-        a = np.array([[1, 1], [2], [3]])
+        a = self._ragged_creation([1, 2, [3]], dtype)
         assert_equal(a.shape, (3,))
         assert_equal(a.dtype, object)
 
-        a = np.array([[1], [2, 2], [3]])
+    @pytest.mark.parametrize('dtype', [object, None])
+    def test_ragged_shape_object(self, dtype):
+        # The ragged dimension of a list is turned into an object array
+        a = self._ragged_creation([[1, 1], [2], [3]], dtype)
```

Comment:
`dtype` appears to be unused above? So its always object I think?

Reply:
`dtype` is used in `_ragged_creation`, once as `object` and once as `None`. The `None` use triggers a `DeprecationWarning`

Reply:
Yes, but `_ragged_creation` always does both, so there is no need for it here.

Reply:
fixing